### PR TITLE
psp-3827 ensure that adding a property to a file from the map populat…

### PR DESCRIPTION
**Pull request by devinleighsmith**
…es all property attributes.

### CODE REVIEW

File: frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx [hunk 1]
```diff
@@ -33,7 +62,7 @@ export const AddAcquisitionContainer: React.FC<IAddAcquisitionContainerProps> =
     history.replace(`/mapview/sidebar/acquisition/${acqFile.id}`);
   };
 
-  const helper = useAddAcquisitionFormManagement({ onSuccess });
+  const helper = useAddAcquisitionFormManagement({ onSuccess, initialForm });
```
Comment (FuriousLlama) [inline frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx:65]: this name does not give much information. Can the logic from the hook be put here or is reused somewhere else?

File: frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx [hunk 2]
```diff
@@ -21,6 +24,32 @@ export const AddAcquisitionContainer: React.FC<IAddAcquisitionContainerProps> =
   const formikRef = useRef<FormikProps<AcquisitionForm>>(null);
 
   const close = useCallback(() => onClose && onClose(), [onClose]);
+  const { selectedResearchFeature, setSelectedResearchFeature } = React.useContext(
+    SelectedPropertyContext,
+  );
+  console.log(selectedResearchFeature);
```
Comment (FuriousLlama) [inline frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx:30]: loggerino

File: frontend/src/components/maps/leaflet/LayerPopup/constants/index.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { pidFormatter } from 'utils';
 export const MUNICIPALITY_LAYER_URL =
   'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
 export const PARCELS_LAYER_URL =
-  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';
+  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_FA_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_FA_SVW';
```
Comment (FuriousLlama) [inline frontend/src/components/maps/leaflet/LayerPopup/constants/index.tsx:6]: is this change related to the fix?

File: frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx (hunk 1 above)
Comment (asanchezr) [reply frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx:65]: This comes down to developer preference. I like to "hide" business logic in forms in single-use hooks. Looks cleaner to me rather than inlining 20 lines of code but I'm open to hear if others dislike that approach and we can decide as a team what we prefer. These hooks are not meant to be reused but rather to hide complexity from a rather large form. I try to think of them as "helpers" or static classes in C#

File: frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx (hunk 2 above)
Comment (asanchezr) [reply frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx:30]: This might also be copy pasta from research file as this form is all about Acquisition File

File: frontend/src/components/maps/leaflet/LayerPopup/constants/index.tsx (hunk 3 above)
Comment (asanchezr) [reply frontend/src/components/maps/leaflet/LayerPopup/constants/index.tsx:6]: I know we wanted to use the recently published FA parcel layer for all layer queries but would think that is a new feature request, not a bug? Unless relevant to this PR

File: frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx [hunk 4]
```diff
@@ -21,6 +24,32 @@ export const AddAcquisitionContainer: React.FC<IAddAcquisitionContainerProps> =
...
+
+  const initialForm = useMemo(() => {
+    const researchForm = new AcquisitionForm();
```
Comment (asanchezr) [inline frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.tsx:33]: seems copy pasta. please rename to acquisitionForm

File: frontend/src/features/properties/selector/components/MapClickMonitor.tsx [hunk 5]
```diff
@@ -46,7 +46,7 @@ export const mapFeatureToProperty = (
     longitude: selectedFeature?.properties?.CLICK_LAT_LNG?.lng ?? latLng.lng ?? undefined,
     planNumber: selectedFeature?.properties?.PLAN_NUMBER ?? undefined,
     address: 'placeholder', //todo: need alternate source for this
-    legalDescription: 'placeholder', //todo: need access to fully attributed parcelmap bc layer,
+    legalDescription: selectedFeature?.properties?.LEGAL_DESCRIPTION,
```
Comment (asanchezr) [inline frontend/src/features/properties/selector/components/MapClickMonitor.tsx:49]: Ah I see why you changed the parcel map URL earlier. We do need FA parcel map to get legal description

Comment (asanchezr) [reply frontend/src/features/properties/selector/components/MapClickMonitor.tsx:49]: Probably relevant to alert testers of the change in map layer so they run regression against everything that uses parcel map layer queries...

File: frontend/src/features/properties/selector/map/PropertyMapSelectorFormView.tsx [hunk 6]
```diff
@@ -5,24 +5,33 @@ import {
...
-    undefined,
+    initialSelectedProperty,
   );
 
+  useEffect(() => {
+    if (initialSelectedProperty !== undefined) {
+      setSelectedProperty(initialSelectedProperty);
+    }
+  }, [initialSelectedProperty]);
+
```
Comment (asanchezr) [inline frontend/src/features/properties/selector/map/PropertyMapSelectorFormView.tsx:34]: question - why do we need this effect here? Wouldn't **selectedProperty** always take the value of **prop.initialSelectedProperty** ?

**Review by asanchezr — APPROVED** [review]

File: frontend/src/components/maps/leaflet/LayerPopup/constants/index.tsx (hunk 3 above)
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/LayerPopup/constants/index.tsx:6]: in order to populate the data from the map, we need to use the FA layer instead of the parcel layer. They both have similar performance and are publicly available so we don't lose any information by making this change.
